src/formatting/formatter.rs

```diff
@@ -386,27 +405,88 @@ impl Formatter for FixFormatter {
                 // unit,
                 // icon,
             } => Err(Error::new_format("'fix' formatter is not implemented yet")),
-            Value::Text(_) => Err(Error::new_format(
-                "Text cannot be formatted with 'fix' formatter",
-            )),
-            Value::Icon(_) => Err(Error::new_format(
-                "An icon cannot be formatted with 'fix' formatter",
-            )),
-            Value::Flag => Err(Error::new_format(
-                "A flag cannot be formatted with 'fix' formatter",
-            )),
+            other => Err(Error::new_format(format!(
+                "{} cannot be formatted with 'fix' formatter",
+                other.type_name(),
+            )))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct DatetimeFormatter {
+    items: Vec<Item<'static>>,
+}
+
+fn make_static_item(item: Item<'_>) -> Item<'static> {
+    match item {
+        Item::Literal(str) => Item::OwnedLiteral(str.into()),
+        Item::OwnedLiteral(boxed) => Item::OwnedLiteral(boxed),
+        Item::Space(str) => Item::OwnedSpace(str.into()),
+        Item::OwnedSpace(boxed) => Item::OwnedSpace(boxed),
+        Item::Numeric(numeric, pad) => Item::Numeric(numeric, pad),
+        Item::Fixed(fixed) => Item::Fixed(fixed),
+        Item::Error => Item::Error,
+    }
+}
+
+impl DatetimeFormatter {
+    fn new(format: String, locale: Option<String>) -> Result<Self> {
+        let items = match locale.as_deref() {
+            Some(locale) => {
+                let locale = locale.try_into().ok().error("invalid locale")?;
+                StrftimeItems::new_with_locale(&format, locale)
+            }
+            None => StrftimeItems::new(&format),
+        }
+        .map(make_static_item)
+        .collect();
+
+        Ok(Self { items })
+    }
+
+    fn _format(&self, val: &Value) -> Result<String> {
+        match val {
+            Value::Datetime(datetime, timezone) => Ok(match timezone {
+                Some(tz) => datetime
+                    .with_timezone(tz)
+                    .format_with_items(self.items.iter())
+                    .to_string(),
+                None => datetime
+                    .with_timezone(&Local)
+                    .format_with_items(self.items.iter())
+                    .to_string(),
+            }),
+            other => Err(Error::new_format(format!(
+                "{} cannot be formatted with 'datetime' formatter",
+                other.type_name(),
+            ))),
         }
     }
 }
 
+impl Formatter for DatetimeFormatter {
+    fn format(&self, val: &Value) -> Result<String> {
+        self._format(val)
+    }
+}
+
+impl Formatter for once_cell::sync::Lazy<DatetimeFormatter> {
```

Comment:
I couldn't find a way around this :(

Reply:
You can convert `&Lazy<T>` to `&T` when choosing the default formatter:

```diff
diff --git a/src/formatting/value.rs b/src/formatting/value.rs
index 11b89fcd..e86e1049 100644
--- a/src/formatting/value.rs
+++ b/src/formatting/value.rs
@@ -127,7 +127,7 @@ impl Value {
         match &self.inner {
             ValueInner::Text(_) | ValueInner::Icon(_) => &formatter::DEFAULT_STRING_FORMATTER,
             ValueInner::Number { .. } => &formatter::DEFAULT_NUMBER_FORMATTER,
-            ValueInner::Datetime { .. } => &formatter::DEFAULT_DATETIME_FORMATTER,
+            ValueInner::Datetime { .. } => &*formatter::DEFAULT_DATETIME_FORMATTER,
             ValueInner::Flag => &formatter::DEFAULT_FLAG_FORMATTER,
         }
     }
```